app/tests/test_search.py

```diff
@@ -1655,16 +1730,108 @@ def test_search_transferring_body_with_search_term_edge_case_path(
 
         assert response.status_code == 200
         soup = BeautifulSoup(response.data, "html.parser")
-        select = soup.find("select", {"id": "sort"})
 
+        select = soup.find("select", {"id": "sort"})
         option_selected = select.find("option", selected=True)
         option_first = select.find("option")
         option_first_value = option_first.get("value")
 
+        decompose_desktop_invisible_elements(soup)
+        inner_table = soup.find("table", {"id": "inner-table"})
+        inner_table_body = inner_table.find("tbody")
+        inner_table_cell_values = get_table_rows_cell_values(inner_table_body)
+
         assert select
         assert option_selected is None
         assert option_first_value == expected_sort_select_value
-        assert evaluate_table_body_rows(soup, expected_results)
+        assert inner_table_cell_values == expected_cell_values
+
+    @pytest.mark.parametrize(
+        "query_params, mock_open_search_return, expected_cell_values",
+        [
+            (
+                "&query=foobar",
+                os_mock_return_tb,
+                [
+                    [
+                        "Test field 1 +1",
+                        "test1 and ... this is just a sentence with a mark element in it",
+                    ],
+                    [
+                        "Test field 2",
+                        "this is a cool test and ... sea shells on the sea shore",
+                    ],
+                ],
+            ),
+            (
+                "&query=foobar&open_all=open_all",
+                os_mock_return_tb,
+                [
+                    [
+                        "Test field 1 +1",
+                        "test1 and ... this is just a sentence with a mark element in it",
+                    ],
+                    [
+                        "Test field 2",
+                        "this is a cool test and ... sea shells on the sea shore",
+                    ],
+                ],
+            ),
+            (
+                "&query=!!!!!",
+                os_mock_return_tb,
+                [
+                    [
+                        "Test field 1 +1",
+                        "test1 and ... this is just a sentence with a mark element in it",
```

Comment:
@anthonyhashemi the way BeautifulSoup strips the text & the tags inside certain elements makes keeping the tags difficult, which is why I've opted to test raw text like this (which is the output we get from BeautifulSoup) and then add the tests to check if the text inside the mark elements is correct further on in the file

Reply:
Oh I see, what elements does it strip and is it a certain method we call on the original soup object that does that? It would be nice to have some test that shows that we persist the marks around the matching text, this test wouldnt catch a bug if we accidentally stripped the tags some how.

Reply:
@anthonyhashemi we gather td elements from the table and for each we call get_text(strip=True), which essentially strips all tags, new lines, etc. and returns the raw text of all elements inside - this is done because some fields may not just contain text, some (like the first cell in the table that also contains the <details> element for triggering the accordion) have other tags inside, these could also be span, divs, etc., perhaps there is a way to retain these tags but at that point we'd also be testing against what is essentially just HTML, which is not ideal because it may change